### DISCUSSION TITLE
地图重写在 iOS + QuantumultX 中无法使用

### DISCUSSION THREAD

**Discussion by Tianket** [NSRingo/.github#25] [Q&A 问答] [answered]
## 系统版本 & 环境

iPhone: iOS 18.2 测试版，国行设备，地区为US，语言为简体中文。已开飞行模式

iPad: iPadOS 17.7 正式版，国行设备，地区为US，语言为简体中文

QuantumultX: 1.5.0-830，开启分流模式

1. 两个设备均使用 QuantumultX 同一配置文件，使用 iCloud 同步。配置文件改自于墨鱼配置
2. 未使用`BoxJs`
3. 地区重写已安装但未开启，`🌤 WeatherKit` 已开启
4. `-*.apple.com` 已移除，但有`-weather-data.apple.com`项

### iPhone端：

1. **无**四处看看(Look Around)功能
2. 高德地图 logo **仍然存在**
3. 将地图拉至最远，**能**显示完整的地球球体模型及昼夜交替的卫星图像
5. 看主要城市地图，右上角显示3D按钮，**不能**查看卫星图像
6. 使用`📍 重写定位`服务，并将地区改为US时，可以正常使用海外全套

### iPad端：

1. **有**四处看看(Look Around)功能，但点开为纯黑色，能看见脚底下的路名
2. 高德地图 logo **仍然存在**
3. 将地图拉至最远，**不能**显示完整的地球球体模型及昼夜交替的卫星图像
4. 看主要城市地图，右上角显示3D按钮，**不能**查看卫星图像

## 报错信息 & Log

见附件。双端均出现了`TypeError`的报错
[log-iphone.txt](https://github.com/user-attachments/files/18083636/log-iphone.txt)
[log-ipad.txt](https://github.com/user-attachments/files/18083635/log-ipad.txt)

***

感谢dalao们的帮助！

**Comment by Tianket** [comment]
如果出现 log 中出现信息泄露（应该是没有）或者 log 没有截全，或者需要更多信息的，烦请dalao联系我

**Comment by VirgilClyne** [comment, marked as answer]
`Quantumult X` 会在脚本未执行完成的情况下，提前清除掉占用大内存的对象
导致您日志中出现脚本运行到一半时对象不存在的情况出现
我们猜测这是 `Quantumult X` 的某种保护机制或内存策略导致的
到目前为止 `Quantumult X` 开发者并未给出答复
请 `Quantumult X` 用户使用 [v4.0.9](https://github.com/NSRingo/GeoServices/releases/tag/v4.0.9) 版
注意，此版本不能将`苹果版`地图整合至`高德版`地图，所以`[动态配置] 资源清单的国家或地区代码`只能选择`🇨🇳中国大陆`

**Reply by Tianket** [reply]
谢谢您的解答。替换为 v4.0.9 版本后，在 `Quantumult X` 的 `引用资源-重写` 页面的本条目中显示 

> 1个问题存在于子资源

随后发现是 v4.0.9 版本会在匹配 `gspe1-ssl` 这条重写规则时，会自动引入本库中 latest 的 `response.js`，但在 latest 中已经不存在这个资源，导致404。

我就不折腾了，坐等 `Quantumult X` 开发者更新了 TAT

再次感谢您的解答，~~薯条姐姐赛高~~

**Reply by VirgilClyne** [reply]
你是不是傻
Location 用最新版
Maps 用 v4.0.9

**Reply by Tianket** [reply]
啊 我的我的 现在好啦 薯条姐姐赛高！！
吐槽一下，国际版的WGS-84坐标真的很不准，定位偏了大概两三公里

**Comment by shzxm** [comment]
也就是说，在目前来使用的话
地图模块依旧使用[v4.0.9](https://github.com/NSRingo/GeoServices/releases/tag/v4.0.9)这个版本吗？
定位模块使用最新的就行？
ios18.5 beta4

**Reply by shzxm** [reply]
尝试了，依旧不管用
quantumult X  beta 1.5.2（844）
